lib/rules/arrow-body-style.js

```diff
@@ -245,6 +257,15 @@ module.exports = {
         }
 
         return {
+            "BinaryExpression[operator='in'] > *:exit"() {
+                isInOp = false;
+            },
+            "BinaryExpression[operator='in']:exit"() {
+                isInOp = true;
+            },
+            "ArrowFunctionExpression"() {
+                isInOp = false;
+            },
```

Comment:
This:

```js
/*eslint arrow-body-style: ["error", "as-needed"]*/

for (var foo = () => { return a in b ? bar : () => {} } ;;);
```

would be auto-fixed to:

```js
/*eslint arrow-body-style: ["error", "as-needed"]*/

for (var foo = () => a in b ? bar : () => {} ;;);
```

which is a syntax error.

Maybe we could try with [this approach](https://github.com/eslint/eslint/pull/13228/files#r443803994)?